web/server.js

```diff
@@ -102,8 +102,73 @@ module.exports = function(options) {
     {
       method: 'POST',
       path: '/login/oauth/access_token',
+      config: {
+        pre: [
+          {
+            assign: 'client',
+            method: function(request, reply) {
+              oauthDb.getClient(request.payload.client_id, function(err, client) {
+                if ( err ) {
+                  return reply(err);
+                }
+                if ( client.client_secret !== request.payload.client_secret ) {
+                  return reply(Boom.forbidden('Invalid Client Credentials'));
+                }
+                reply(client);
+              });
+            }
+          },
+          {
+            assign: 'redirectUri',
```

Comment:
I was actually speccing out that we ignore any application-submitted redirect url entirely, and only use the registered one